src/time.rs

```diff
@@ -0,0 +1,197 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use pyo3::exceptions::PyValueError;
+use pyo3::prelude::*;
+use pyo3::pyclass::CompareOp;
+use pyo3::types::PyType;
+use pyo3::PyResult;
+
+use crate::binding_utils::hash_generic;
+
+#[pyfunction]
+pub(crate) fn mock_time(time: &PyAny) -> PyResult<()> {
+    use libparsec::types::MockedTime::*;
+    libparsec::types::DateTime::mock_time(if let Ok(dt) = time.extract::<DateTime>() {
+        FrozenTime(dt.0)
+    } else if let Ok(dt) = time.extract::<i64>() {
+        ShiftedTime(dt)
+    } else if time.is_none() {
+        RealTime
+    } else {
+        return Err(PyValueError::new_err("Invalid field time"));
+    });
+    Ok(())
+}
+
+#[pyclass]
+#[derive(PartialEq, Eq, Clone)]
+pub(crate) struct DateTime(pub libparsec::types::DateTime);
+
+#[pymethods]
+impl DateTime {
```

Comment:
That's a duplication of the binding right? What's the point of having this in `libparsec_python` too?

Reply:
Because needed by other files